homeassistant/components/nest/camera_sdm.py

```diff
@@ -208,11 +208,13 @@ async def async_camera_image(
         stream = await self.async_create_stream()
         if stream:
             return await stream.async_get_image(width, height)
-        if not self._placeholder_image:
-            self._placeholder_image = await self.hass.async_add_executor_job(
-                PLACEHOLDER.read_bytes
-            )
-        return self._placeholder_image
+        return await self.hass.async_add_executor_job(self.placeholder_image)
+
+    @classmethod
+    @functools.lru_cache(maxsize=1)
```

Comment:
What's the benefit of setting the `maxsize` here? Since the method gets called with the same argument (`cls`) every time, won't it just store one value no matter how many times it's called? If we don't need to use `maxsize`, then we can also consider using `cache` or `cached_property` instead.
(BTW I didn't know about `cached_property` in functools - I thought you were talking about this https://pypi.org/project/cached-property/ instead which is why I brought up that ttl stuff in the previous PR.)

Reply:
I had a feeling these didn't work with class methods, and also I got the impression we don't like I/o happening in properties. I agree they shave the semantics I want but I didn't think it would actually work.

Reply:
I think avoiding I/O in properties is so that no hidden I/O happens when accessing stuff especially from other components. If we make sure we are only using it internally and make it "private" maybe it's ok. But you're right, maybe it makes sense to avoid it.
But as for the first question I asked: what's the use of `maxsize` here? If we determine we don't need `maxsize` then we can just use `cache` as it's the same thing as `lru_cache(maxsize=None)`: https://docs.python.org/3/library/functools.html#functools.cache

Reply:
Fixed to use cache.